### PR TITLE
gBoards FaunchPad

### PR DESCRIPTION
**Pull request by germ**
Part two of the gboards-consolidation project. The Mega PR has been separated out into smaller per-keyboard PRs.

This PR contains a single keyboard as per @zvecr , in this case FaunchPad. This code was pulled from #8262 

- [x] Enhancement/optimization
- [x] Keymap/layout/userspace (addition or update)

## Issues Fixed or Closed by This PR

- Replaces #8262
- Waiting on #8921

### CODE REVIEW

File: keyboards/gboards/k/faunchpad/config.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x1337
```
[the remark this answers is not in the document]
Comment (germ) [reply keyboards/gboards/k/faunchpad/config.h]: Fixed in https://github.com/qmk/qmk_firmware/pull/8925/commits/3f802f9fb805b69deee287a472d05fb0d4e24d02

**Comment by germ** [comment]
What's holding this back?

**Comment by Erovia** [comment]
> What's holding this back?

Mostly the lack of resources to review PRs.

But from a quick glance, I'd recommend the same things as I did for #8928.

* modernize pin initialization

* check if custom matrix is really needed, if yes, check if `custom matrix lite` is enough

Btw, this is a new (to QMK) board, right?

**Comment by germ** [comment]
Makes sense, I'll see what I can do about the updated interface but the main reason I'm avoid it is simply lack of time due to COVID related backlog and I need to refactor/test 9 boards. :P

This is a new board though, same with Ginny. The Split boards operate just like the ergoDox, so anything that would work for it will work for my splits

**Comment by Erovia** [comment]
No worries, somehow life is quite busy for everybody right now, I wonder why.... :D

Since this is a new board, I removed the breaking_change tag, so that's at least -1 concern.

**Comment by germ** [comment]
As far as custom matrix goes, by the looks of it I'll need that (same as the EZ guys). I can work on migrating to the new lite code when I have a chance (and that will help clean up the code a bunch). 

As far as the other recommendations I'll try and get those in soon

File: keyboards/gboards/faunchpad/config.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x0007
+#define PRODUCT_ID 0x0006
+#define DEVICE_VER 1
```
Comment (noroadsleft) [inline keyboards/gboards/faunchpad/config.h:8]:
```suggestion
#define DEVICE_VER 0x0001
```

File: keyboards/gboards/faunchpad/config.h [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+#define MANUFACTURER g Heavy Industries
+#define PRODUCT FaunchPad
+#define DESCRIPTION Teeny Toiny Macropad
```
Comment (noroadsleft) [inline keyboards/gboards/faunchpad/config.h:11]:
```suggestion
```

File: keyboards/gboards/faunchpad/faunchpad.h [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_faunch(k00, k01, k02, k03, k10, k11, k12, k13) \
+    { {k00, k01, k02, k03}, {k10, k11, k12, k13}, }
```
Comment (noroadsleft) [inline keyboards/gboards/faunchpad/faunchpad.h:6]:
```suggestion
#define LAYOUT_ortho_2x4( \
    k00, k01, k02, k03, \
    k10, k11, k12, k13  \
) { \
    { k00, k01, k02, k03 }, \
    { k10, k11, k12, k13 }  \
}
```

File: keyboards/gboards/faunchpad/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ *	| FP01 | FP02 | FP03 | FP04 | 
+ *  |---------------------------|
+ *	| FP11 | FP12 | FP13 | FP14 | 
+ *  '---------------------------'
+ */
+
+// Don't edit your keymap here! Do it in your dicts/ !
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_faunch( 	KC_1,  KC_2,  KC_3,  KC_4,
+  										  KC_5,  KC_6,  KC_7,  KC_8)
```
Comment (noroadsleft) [inline keyboards/gboards/faunchpad/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_ortho_2x4(
        KC_1,  KC_2,  KC_3,  KC_4,
        KC_5,  KC_6,  KC_7,  KC_8
    )
```

File: keyboards/gboards/faunchpad/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT_faunch( 	KC_1,     KC_2,     KC_3,     KC_4,
+  										  KC_5,     KC_6,     KC_7,     KC_8),
+  [1] = LAYOUT_faunch( 	KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+                        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS)
```
Comment (noroadsleft) [inline keyboards/gboards/faunchpad/keymaps/via/keymap.c:6]:
```suggestion
    [0] = LAYOUT_ortho_2x4(
        KC_1,     KC_2,     KC_3,     KC_4,
        KC_5,     KC_6,     KC_7,     KC_8
    ),
    [1] = LAYOUT_ortho_2x4(
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS
    ),
    [2] = LAYOUT_ortho_2x4(
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS
    ),
    [3] = LAYOUT_ortho_2x4(
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
        KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS
    )
```

**Comment by Frenzie** [comment]
Fwiw, works great on my Faunchpad.

**Pull request closed without merging**